src/Teapot.Web/Controllers/StatusController.cs

```diff
@@ -23,28 +24,29 @@ public StatusController(TeapotStatusCodeResults statusCodes)
 
     [Route("{statusCode:int}", Name = "StatusCode")]
     [Route("{statusCode:int}/{*wildcard}", Name = "StatusCodeWildcard")]
-    public IActionResult StatusCode(int statusCode, int? sleep) {
+    public IActionResult StatusCode(int statusCode, int? sleep, bool? suppressBody) {
         var statusData = _statusCodes.ContainsKey(statusCode)
             ? _statusCodes[statusCode]
             : new TeapotStatusCodeResult { Description = $"{statusCode} Unknown Code" };
 
         sleep ??= FindSleepInHeader();
+        suppressBody ??= FindSuppressBodyInHeader();
 
         var customResponseHeaders = HttpContext.Request.Headers
             .Where(header => header.Key.StartsWith(CUSTOM_RESPONSE_HEADER_PREFIX))
             .ToDictionary(header => header.Key.Replace(CUSTOM_RESPONSE_HEADER_PREFIX, string.Empty), header => header.Value);
 
-        return new CustomHttpStatusCodeResult(statusCode, statusData, sleep, customResponseHeaders);
+        return new CustomHttpStatusCodeResult(statusCode, statusData, sleep, suppressBody, customResponseHeaders);
     }
 
     [Route("Random/{range?}", Name = "Random")]
     [Route("Random/{range?}/{*wildcard}", Name = "RandomWildcard")]
-    public IActionResult Random(string range = "100-599", int? sleep = null)
+    public IActionResult Random(string range = "100-599", int? sleep = null, bool? suppressBody = null)
     {
         try
         {
             var statusCode = GetRandomStatus(range);
-            return StatusCode(statusCode, sleep);
+            return StatusCode(statusCode, sleep, suppressBody);

```

Comment:
Shouldn't it also use the `FindSuppressBodyInHeader` in this action like it does in `StatusCode`?

Reply:
Although I do note that we don't do that for `sleep` so you were probably following the pattern that was already not properly done 😅

Reply:
If you want me to update both, can do.

Reply:
Actually scratch that, it's actually calling the `StatusCode` **action**, not the `StatusCode` method on the `Controller` base class. Super confusing but that was already the case. So, `FindSuppressBodyInHeader` **is** being called.

Reply:
Oh yes, good point. I really should know how this code works 🤣